components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1011,27 +1013,10 @@ HDKeyring* KeyringService::RestoreKeyring(mojom::KeyringId keyring_id,
   return keyring;
 }
 
-mojom::KeyringInfoPtr KeyringService::GetKeyringInfoSync(
-    mojom::KeyringId keyring_id) {
-  DCHECK(profile_prefs_);
-  mojom::KeyringInfoPtr keyring_info = mojom::KeyringInfo::New();
-  keyring_info->id = keyring_id;
-  keyring_info->is_keyring_created = IsKeyringCreated(keyring_id);
-  keyring_info->is_locked =
-      !keyring_info->is_keyring_created || IsLocked(keyring_id);
-  bool backup_complete = false;
-  const base::Value* value =
-      GetPrefForKeyring(*profile_prefs_, kBackupComplete, keyring_id);
-  if (value) {
-    backup_complete = value->GetBool();
-  }
-  keyring_info->is_backed_up = backup_complete;
-  return keyring_info;
-}
-
-void KeyringService::GetKeyringInfo(mojom::KeyringId keyring_id,
-                                    GetKeyringInfoCallback callback) {
-  std::move(callback).Run(GetKeyringInfoSync(keyring_id));
+bool KeyringService::IsWalletSetup() {
```

Comment:
Could we expose this in mojom? Otherwise we'd likely implement the same helper on iOS

Reply:
Android and desktop are ok without this, why iOS needs it?
I mean I surely could expose it with mojom but I want to avoid wallet's frontends doing same things differently.

Reply:
We were using KeyringInfo's `is_keyring_created` to determine if we show onboarding on iOS. Android [seems to be using a boolean flag in SharedPreferences](https://github.com/brave/brave-core/blob/e03236d0e5d709a258d47179f2928b9b2275c41b/android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java#L219-L222) but I don't see how desktop is determining this, is there a better method now with KeyringInfo / `is_keyring_created` removed?

Reply:
Ok, fair point! I need to think now what would be the best approach here

Reply:
https://github.com/brave/brave-core/pull/20488/commits/73f2b4fa03f0ea1f86b2f652867e296a7dbb98f4

Reply:
Thanks!